javascript/javascript3/week3/main.js

```diff
@@ -0,0 +1,93 @@
+const urlForm = document.getElementById('urlForm');
+const screenshotPreview = document.getElementById('screenshotPreview');
+const saveButton = document.getElementById('saveButton');
+const savedScreenshotsList = document.getElementById('savedScreenshotsList');
+
+urlForm.addEventListener('submit', async (e) => {
+  e.preventDefault();
+  const urlInput = document.getElementById('urlInput');
+  const url = urlInput.value;
+
+  const screenshotUrl = await generateScreenshot(url);
+
+  if (screenshotUrl) {
+    // Display the screenshot
+    screenshotPreview.innerHTML = `<img src="${screenshotUrl}" alt="Screenshot" />`;
+
+    // Enable save button
+    saveButton.disabled = false;
+  } else {
+    screenshotPreview.innerHTML = 'Failed to generate screenshot.';
+  }
+});
+
+saveButton.addEventListener('click', async () => {
+  const urlInput = document.getElementById('urlInput');
+  const url = urlInput.value;
+
+  const screenshotUrl = await generateScreenshot(url);
+
+  if (screenshotUrl) {
+    await saveScreenshot(url, screenshotUrl);
+  } else {
+    console.log('Failed to generate screenshot.');
+  }
+});
+
+const generateScreenshot = async (url) => {
+  const encodedUrl = encodeURIComponent(url);
+  const screenshotUrl = `https://website-screenshot6.p.rapidapi.com/screenshot?url=${encodedUrl}&width=1920&height=1080&fullscreen=true`;
```

Comment:
You don't want to hardcode the screen size since that isn't responsive. 

```javascript
const screenWidth = window.screen.width;
const screenHeight = window.screen.height;
```